### PR TITLE
425 improve wrong type exceptions

### PR DESCRIPTION
**Pull request by dh94**
fixing https://github.com/graphql-java/graphql-java/issues/425

### CODE REVIEW

File: src/main/java/graphql/schema/idl/errors/NotAnOutputTypeError.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package graphql.schema.idl.errors;
+
+import graphql.language.TypeDefinition;
+
+import static java.lang.String.format;
+
+public class NotAnOutputTypeError extends BaseError {
+
+    public NotAnOutputTypeError(TypeDefinition typeDefinition) {
+        super(typeDefinition, format("The %s type is used as an OutputType, but is not declared as one", typeDefinition.getName()));
```
Comment (andimarek) [inline src/main/java/graphql/schema/idl/errors/NotAnOutputTypeError.java]: Maybe we can phrase that a bit diffrent: "expected OutputType, but found %s type" ?

File: src/main/java/graphql/schema/idl/errors/NotAnInputTypeError.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package graphql.schema.idl.errors;
+
+import graphql.language.TypeDefinition;
+
+import static java.lang.String.format;
+
+public class NotAnInputTypeError extends BaseError {
+
+    public NotAnInputTypeError(TypeDefinition typeDefinition) {
+        super(typeDefinition, format("The %s type is used as an InputType, but is not declared as one", typeDefinition.getName()));
```
Comment (andimarek) [inline src/main/java/graphql/schema/idl/errors/NotAnInputTypeError.java]: Maybe we can phrase that a bit diffrent: "expected InputType, but found %s type" ?

**Comment by dh94** [comment]
@andimarek changed message as requested 💃

**Comment by andimarek** [comment]
Thanks!

File: src/main/java/graphql/schema/idl/errors/NotAnInputTypeError.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package graphql.schema.idl.errors;
+
+import graphql.language.TypeDefinition;
+
+import static java.lang.String.format;
+
+public class NotAnInputTypeError extends BaseError {
+
+    public NotAnInputTypeError(TypeDefinition typeDefinition) {
+        super(typeDefinition, format("expected InputType, but found %s type", typeDefinition.getName()));
```
Comment (bbakerman) [inline src/main/java/graphql/schema/idl/errors/NotAnInputTypeError.java]: Have  a look at the other errors.  They try to include a linecol() of the offending definition

Comment (bbakerman) [inline src/main/java/graphql/schema/idl/errors/NotAnInputTypeError.java]:
```
public class MissingInterfaceTypeError extends BaseError {

    public MissingInterfaceTypeError(String typeOfType, TypeDefinition typeDefinition, TypeName typeName) {
        super(typeDefinition, format("The %s type '%s' is not present when resolving type '%s' %s",
                typeOfType, typeName.getName(), typeDefinition.getName(), lineCol(typeDefinition)));
    }
}
```

Comment (andimarek) [inline src/main/java/graphql/schema/idl/errors/NotAnInputTypeError.java]: @bbakerman I merged it already ... my bad.

**Comment by andimarek** [comment]
@dh94 do you mind looking at the latest comment by @bbakerman? He has a good suggestion for  even better error messages ... thanks a lot

**Comment by dh94** [comment]
@andimarek @bbakerman  sure thing, should i create a new pull request for this?

**Comment by andimarek** [comment]
@dh94 yes, that would be great .... thanks!

**Comment by dh94** [comment]
@andimarek  @bbakerman  created https://github.com/graphql-java/graphql-java/pull/437

**Pull request merged**